fetcher/fedora.go

```diff
@@ -0,0 +1,428 @@
+package fetcher
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/xml"
+	"fmt"
+	"net/url"
+	"regexp"
+	"strings"
+
+	"github.com/inconshreveable/log15"
+	"github.com/vulsio/goval-dictionary/util"
+	"golang.org/x/xerrors"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	archX8664   = "x86_64"
+	archAarch64 = "aarch64"
+
+	fedoraUpdateURL = "https://dl.fedoraproject.org/pub/fedora/linux/updates/%s/Everything/%s/repodata/repomd.xml"
+	fedoraModuleURL = "https://dl.fedoraproject.org/pub/fedora/linux/updates/%s/Modular/%s/repodata/repomd.xml"
+	bugZillaURL     = "https://bugzilla.redhat.com/show_bug.cgi?ctype=xml&id=%s"
+	kojiPkgURL      = "https://kojipkgs.fedoraproject.org/packages/%s/%s/%s/files/module/modulemd.%s.txt"
+)
+
+// FetchUpdateInfosFedora fetch OVAL from Fedora
+func FetchUpdateInfosFedora(versions []string) (FedoraUpdatesPerVersion, error) {
+	// map[osVer][updateInfoID]FedoraUpdateInfo
+	uinfos := make(map[string]map[string]FedoraUpdateInfo, len(versions))
+	for _, arch := range []string{archX8664, archAarch64} {
+		reqs, moduleReqs := newFedoraFetchRequests(versions, arch)
+		results, err := fetchEverythingFedora(reqs)
+		if err != nil {
+			return nil, xerrors.Errorf("fetchEverythingFedora. err: %w", err)
+		}
+
+		moduleResults, err := fetchModulesFedora(moduleReqs, arch)
+		if err != nil {
+			return nil, xerrors.Errorf("fetchModulesFedora. err: %w", err)
+		}
+		results.merge(&moduleResults)
+
+		for osVer, result := range results {
+			if _, ok := uinfos[osVer]; !ok {
+				uinfos[osVer] = make(map[string]FedoraUpdateInfo, len(result.UpdateList))
+			}
+			for _, uinfo := range result.UpdateList {
+				if tmp, ok := uinfos[osVer][uinfo.ID]; ok {
+					uinfo.Packages = append(uinfo.Packages, tmp.Packages...)
+					uinfo.Packages = uniquePackages(uinfo.Packages)
```

Comment:
Is it necessary to set it to uinfo.Packages once?
```suggestion
					uinfo.Packages = uniquePackages(append(uinfo.Packages, tmp.Packages...))
```